File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    int start, end;
    start=end=1;
    while(start<=n){
        int sum = 0;
        for(int i=start;i<=end;++i) sum+=i;
        if(sum<n) end++;
        else if(sum>n) start++;
        else answer++, end++;
    }
    return answer;
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    cnt = sn = 0
    s = e = 1
    while s <= n :
        if sn < n :
            sn += e
            e += 1
        else:
            if sn == n :
                cnt += 1
            sn -= s
            s += 1
    return cnt